packages/mobile/src/identity/verification.ts

```diff
@@ -618,13 +623,55 @@ export function attestationCodeReceiver(
       return
     }
 
+    const allIssuers = attestations.map((a) => a.issuer)
+    let securityCode: string | null = null
+    let attestationCode
+
     try {
-      const code = extractAttestationCodeFromMessage(action.message)
-      if (!code) {
+      if (features.SHORT_VERIFICATION_CODES) {
+        securityCode = extractShortSecurityCodeMessage(action.message)
+        if (securityCode) {
+          const lookupHash = parseInt(securityCode[0], 10)
+          const lookupAttestations = attestations.filter(
+            (attestation) => hashAddressToSingleDigit(attestation.issuer) === lookupHash
+          )
+
+          for (const attestation of lookupAttestations) {
+            // Try to recover the full attestation code from the matching issuer's attestation service
+            try {
+              const message = yield call(
+                getAttestationCodeFromSecurityCode,
+                attestationsWrapper,
+                account,
+                phoneHashDetails,
+                attestation,
+                securityCode.substr(1)
+              )
+              attestationCode = extractAttestationCodeFromMessage(message)
+              break
+            } catch (e) {
+              Logger.warn(
+                TAG + '@attestationCodeReceiver',
+                'Getting attestation code for security code error: ',
+                e
+              )
+              continue
+            }
+          }
+        }
+      } else {
+        attestationCode = extractAttestationCodeFromMessage(action.message)
```

Comment:
It might be cleaner if we run this out of the else statement regardless and just change what we pass into it.
```
let message = action.message
if (features.SHORT_VERIFICATION_CODES) {
  ...
  message = yield call(...)
}
attestationCode = extractAttestationCodeFromMessage(message)
```